### PR TITLE
Design updates for SecureDrop 1.0.0, rev1

### LINKED ISSUE #4500: Source UI design updates: color!
## Design Epic 
Design updates to the Source UI (w/in the existing experience) are being explored in [UX-repo issue #42](https://github.com/freedomofpress/securedrop-ux/issues/42). Within these updates, a new color palette is being used. The goal of the color updates are make the product reflect the SecureDrop brand, as it's presented on the website. They also make the product snazzier.

**This issue has been drafted to implement only the items that pertain to color, from the design epic in the aforementioned GH Issue.**

As such, div background colors, border colors, font colors, and hr styles comprise the majority of the below to-do list. A few stretch-goal items here and there, but they are purely aspirational in nature and not required to close the issue.

In a few places there might be a placed PNG that keeps to the old color schema. Thus far, we have identified the bulleted items as closely related issues that should be addressed in follow-up PRs. If we’ve missed anything, please comment below, and we’ll update the below list accordingly! :)
- Default logo the product ships with #4503
- Update icons placed on Submit/Cancel buttons in Source Interface #4501
- Favicon #4502 
 
## This Issue
Update core styles in the Source UI to reflect the below spec’d values. If time permits, a few identified stretch-goals would be extra special. 

### Global
See _Submit_ page mockup:  [static](https://scene.zeplin.io/project/5cb7d4d1f93bd116d7b25561)
Note: Pay no mind to font size, family, or text-decoration stylings in this mockup. Nina's Tor browser fonts load wackily. Likewise, the mockup does not show hyperlink underlines styled w/ CSS.
- [x] Change `H1`, `H2`, and `H3` to all be `#2a319d`
- [x] Put headlines for "Submit files" and "Read replies" on same level (both should be H2)
- [x] Change static hyperlink color to be  <s>`#0075F7`</s> `#0065db` (previous color was seen as too bright), and `#2a319d` on hover
  - [x] Stretch goal: re-style hyperlinks to use bottom-border attribute, with 40% alpha when static and 100% alpha on hover
     - Using CSS to style hyperlink underlines enables subtle things to enhance legibility of text, by nudging the underline a pixel lower than it usually is and by reducing the static alpha value of the border. By doing both, underlines visually interfere with letterform descenders (g, p, y, j each have _descenders_ in lower-case form) less, and legibility is enhanced w/o interfering with link discovery.
- [x] Change total-UI container box outline (the 1px grey box that encloses everything other than the newsroom logo, footer, and languages widget) to be `#e1e1e1` (currently #c3c3c3)
  - The goal is for this to eventually go away, with some other tweaks; but for now, it should be de-emphasized.

### Index Page
See mockups: [static](https://scene.zeplin.io/project/5cb7d4d1f93bd116d7b25561/screen/5ccc9771c60ef46833e628f4) · [button hover](https://scene.zeplin.io/project/5cb7d4d1f93bd116d7b25561/screen/5cf5c209cee3f113e9e06614)

_Note:_ Pay no mind to font size, family, or text-decoration stylings in this mockup. Nina's Tor browser fonts load wackily. Likewise, the mockup does not show hyperlink underlines styled w/ CSS.
- [x] Both `hr`s: change styling from `dashed` to `dotted`
- [x] Stretch goal: move the languages widget beneath the logo, as it is on all the other pages
#### “Submit for the first time” box
- [x] Box fill: `background-image: linear-gradient(140deg, #0096dc, #005db7);`
- [x] `hr`, text, button text color, and `button border`: `white`
- [x] Button on hover: Keep white icon, border=0, button bg   `background-image: linear-gradient(72deg, #2a319d, #2d3273);`
#### “Already submitted something?” box
- [x] Outline:  `border-width: 1.5px; border-image-source: linear-gradient(145deg, #0096dc, #005db7); border-image-slice: 1;`
- [x] Text & `hr`: `#0A7CDC`
- [x] Button color: `background-image: linear-gradient(111deg, #0082d9, #126ec8);` 
- [x] Button on hover: Keep white icon, border=0, button bg   `background-image: linear-gradient(72deg, #2a319d, #2d3273);`
    
### Submit Page
See mockup:  [static](https://scene.zeplin.io/project/5cb7d4d1f93bd116d7b25561)
_Note:_ Button icons to be done in a separate PR. Pay no mind to font size, family, or text-decoration stylings in this mockup. Nina's Tor browser fonts load wackily. Likewise, the mockup does not show hyperlink underlines styled w/ CSS.
#### Upload/Message Box
- [x] Grey box behind cloud icon and “Browse” bar: update background to:  `background-image: linear-gradient(115deg, #f8fbff, #eaf1f8);`
- [ ] Cloud icon to 20% alpha
#### Buttons
_The button icons are tracked in #4501_
- [x] Primary bg color: <s>`#0075F7`</s> `#0065db` (previous color was seen as too bright); on hover, `#2a319d`
- [x] Cancel border color & text:  <s>`#0075F7`</s> `#0065db` (previous color was seen as too bright); on hover, `#2a319d`

### LINKED ISSUE #4501: Update icons placed on Submit/Cancel buttons in Source Interface
## Design Epic 
Design updates to the Source UI (w/in the existing experience) are being explored in [UX-repo issue #42](https://github.com/freedomofpress/securedrop-ux/issues/42). Within these updates, a new color palette is being used. The goal of the color updates are make the product reflect the SecureDrop brand, as it's presented on the website. They also make the product snazzier.

**This issue has been drafted to implement the below in a PR after #4500 has been completed**
 
## This Issue
See mockup:  [static](https://scene.zeplin.io/project/5cb7d4d1f93bd116d7b25561)
- [x] Update "x" icon to `sd-blue` artwork w/o circle in static mode, and `grimace-blue` artwork for hover. Both icons as PNGs are [here](https://drive.google.com/open?id=1ycflbtla_9ZU0ErKBv-pqEUThfCcKl4w).
- [x] Eliminate checkmark icon on `Submit`
   - This is very confusing, as it indicates an action has already happened; when a file has been selected to be uploaded but not yet submitted, it's more confusing. :(

### LINKED ISSUE #4502: Update favicon: color!
## Design Epic 
Design updates to the Source UI (w/in the existing experience) are being explored in [UX-repo issue #42](https://github.com/freedomofpress/securedrop-ux/issues/42). Within these updates, a new color palette is being used. The goal of the color updates are make the product reflect the SecureDrop brand, as it's presented on the website. They also make the product snazzier.

**This issue has been drafted to implement the below in a PR after #4500 has been completed**
 
## This Issue
- [x] Update SecureDrop favicon that ships with the product. `sd_favicon-2019.png` is [here](https://drive.google.com/open?id=1ycflbtla_9ZU0ErKBv-pqEUThfCcKl4w)

### LINKED ISSUE #4503: Default logo update: color!
## Design Epic 
Design updates to the Source UI (w/in the existing experience) are being explored in [UX-repo issue #42](https://github.com/freedomofpress/securedrop-ux/issues/42). Within these updates, a new color palette is being used. The goal of the color updates are make the product reflect the SecureDrop brand, as it's presented on the website. They also make the product snazzier.

**This issue has been drafted to implement the below in a PR after #4500 has been completed**
 
## This Issue
- [x] Update default newsroom logo (the SecureDrop logo) that ships with the product. `sd_logo-2019.png` is [here](https://drive.google.com/open?id=1ycflbtla_9ZU0ErKBv-pqEUThfCcKl4w)

### PR DESCRIPTION
**Pull request by eloquence**
Updates colors throughout to a new "SecureDrop Blue" and a darker
"Grimace Blue", a new "danger" color, and blue gradients on the
index page; updates the favicon and the default logo.

Resolves #4500
Resolves #4501
Resolves #4502
Resolves #4503

Additional tweaks chiefly agreed upon with Nina Alter (@ninavizz)
during pair development:

Global:
- Removal of some button icons that did not disambiguate meaning
- Tweaks to default button letter spacing, font weight
- Pick headline color that fits with new overall theme
- Ensure consistent use of headline levels
- Removed unused colors, styles, icons
- More careful spacing between icons and text on buttons

Source Interface:
- New trashcan icon for deleting replies
- Flipped order of introductory text on submit page by importance

Journalist Interface:
- Restyled timestamps, "0 docs, 1 message" type counters to be less easily confused with buttons
- Restyled "Select all" type buttons to not have checkbox icons on them, to ensure they're not confused with actual checkboxes
- Added spacing between Select buttons and other actions, to make logical distinction more obvious
- Removed download icon from each table row, as it did not convey sufficient meaning to justify keeping it
- Removed "Reply" icon from Reply headline, and left-aligned it, for consistency with other headlines
- Made textbox the same width as the table above it, and removed fixed column width, to ensure proper scaling across viewport sizes
- Right-aligned "Submit" button consistent with other forms
- Removed hover from "X" of modals, as it did not really add value here

Admin Interface:
- Made it more obvious when the Delete icon is disabled

Out of scope for this rev:
- String changes
- Typography changes
- Docs screenshot updates (will be done after final rev)

## Test plan

- Check the changes against the changelog above.
- Verify all paths through the Source Interface, including:
   - submitting a file or message
   - changing the UI language
   - deleting a reply from a journalist
- Verify all paths through the Journalist Interface, including:
   - "Flag for reply" flow
   - replying to a source
   - deletion of collections and sources
   - deletion of replies by sources
   - changing codename of a source
   - adding a user
   - other administrative functions

When stepping through these paths, make note of any obvious design regressions, or design changes which you are uncertain about ("is it intentional that the hover behavior between these two buttons is inconsistent", etc.).

## Deployment considerations

- Screenshots in docs will have to be updated after final rev
- We'll reach out to news organizations prior to 1.0.0 to keep them informed about the logo change, so they can account for it on landing pages and in other materials
- We'll prominently explain the UI changes in the changelog for 1.0.0

## Screenshots

Screen Shot 2019-07-30 at 17 35 48-fullpage
Screen Shot 2019-07-30 at 17 35 50-fullpage
Screen Shot 2019-07-30 at 17 36 13-fullpage
Screen Shot 2019-07-30 at 17 36 26-fullpage
Screen Shot 2019-07-30 at 17 36 33-fullpage
Screen Shot 2019-07-30 at 17 37 06-fullpage
Screen Shot 2019-07-30 at 17 37 14-fullpage
Screen Shot 2019-07-30 at 17 37 49-fullpage
Screen Shot 2019-07-30 at 17 38 06-fullpage

### CODE REVIEW

**Comment by eloquence** [comment]
Asks for @ninavizz: 
- Pls provide larger version of "X" icon as noted in #4501
- Pls review favicon on dark background in #4502 - more tweaks required?
- Pls review Journalist UI color scheme, and recommendation for either decoupled or unified color scheme
- Pls point out any other issues you notice w/ current draft impl 

Huge thanks for getting us this far, really looking forward to doing this refresh as part of 1.0.0 :)

File: securedrop/sass/_source_index.sass [hunk 1]
```diff
@@ -31,22 +36,55 @@
...
       border-style: none
 
-    .index-top
-      border-top-width: 1px
+    .index-right
+      background: white
+      border-width: 1.5px
+      border-image-source: linear-gradient(145deg, #0096dc, #005db7)
+      border-image-slice: 1
+      border: 1px solid
```
Comment (rmol) [inline securedrop/sass/_source_index.sass]: I think you need to remove `border: 1px solid` here so it doesn't override the gradient border.

**Comment by rmol** [comment]
Took a spin through the changes to check the source interface against the four issues listed, and they look pretty good. 

Made a comment about the one bug I found, preventing the border image gradient on the source index page. 

The only other thing I noticed was that setting the alpha to 20% for the cloud icon on the submit page would make it too light, IMO -- but that item is as yet unchecked on the issue page, so maybe it's not yet nailed down.

File: securedrop/sass/_source_index.sass (hunk 1 above)
Comment (eloquence) [reply securedrop/sass/_source_index.sass]: Good catch, thanks! (Should be fixed.)

**Comment by eloquence** [comment]
@ninavizz and I made a fair number of smaller tweaks as we audited the colors and styles used throughout the Journalist and Source UI in the last few days. I've attempted to provide a complete changelog in the PR and in the commit summary.

To manage scope, I'm going to call this rev1; at this point, updates to this PR will be to identify issues flagged during review. We'll be discussing the scope of a potential rev2 in the next UX meeting, on August 1.

**Comment by rmol** [comment]
:+1: LGTM. The border bug is fixed, and the test changes are good.

**Comment by eloquence** [comment]
@ninavizz Just a note that the full-size logo in this PR still uses the old keyhole cut; could you provide me with a 500 x 449 resolution transparent PNG version with the revised cut?

**Comment by eloquence** [comment]
(Rebased.)

File: securedrop/sass/_source_index.sass [hunk 2]
```diff
@@ -31,22 +36,60 @@
...
+      flex-direction: column
+      justify-content: space-between
       width: 48%
-      border: 1px solid $color_purple_medium
       border-top-width: 0
       border-bottom-width: 0
 
     .index-left
-      background: $color_purple_medium
+      background-image: linear-gradient(140deg, #0096dc, #005db7)
```
Comment (emkll) [inline securedrop/sass/_source_index.sass]: would using variables for theses colors make sense here, since they are reused later?

**Comment by eloquence** [comment]
Thanks for the comments @emkll! You're right that those colors are still not ideal. We kept them out of scope for now to keep scope constrained, but they're easy to change, and your comment encouraged us to poke a little. Nina and I will sync up on that tomorrow and may make some final tweaks here.

**Comment by ninavizz** [comment]
Thank you for flagging the design concerns @emkll! I agree with all of the points you've made.

The _Get Codename_ page and all Messaging components need more love than just color tweeks, unfortunately. Adjusting the colors, alone, with the visual language currently used in both would potentially make those parts of the UI look "more broken," than simply untouched by the current round of UI updates. If that makes sense. 

I'll defer to @eloquence on how to scope agreed to updates across which PRs; this one (`Rev1`) and a `Rev2` are being created for 1.0.0. Per your mention in your review, however, Erik and I did just agree to extend the scope of the `Rev2` items to include _Get Codename_ page updates... and possibly an update to the _Codename Widget_ on the _Submit_ page. Both of which, are certainly needed.

Messaging I'd like to revisit holistically in either a point release, or in 1.1.0. At that time colors, iconography, styling, and text will all be overhauled. The Tor messaging banner(s) on the `index` page, tho, I do consider to be outside the scope of *that,* as I consider it to be its own design problem. We also have yet to get clarification on the exact urgency of that ask—which will factor heavily into which direction we take in that design solution. Not sure if you've got that on your own radar atm to look into, or if I should create a new GH Issue for that? Pls advise. 

TL;DR, this PR I'm kosher thumbs-up'ing, with the caveat we get the flagged _Get Codename_ updates into the Rev2 PR.

File: securedrop/journalist_templates/edit_account.html [hunk 3]
```diff
@@ -72,7 +72,7 @@ <h2>{{ gettext('Reset Password') }}</h2>
     <br>
     <span id="password" class="password">{{ password }}</span>
   </p>
-  <button class="sd-button btn" type="submit" id="reset-password"><i class="fas fa-sync"></i> {{ gettext('Reset Password') }}</button>
+  <button type="submit" id="reset-password"><i class="fas fa-sync"></i> {% filter upper %}{{ gettext('Reset Password') }}{% endfilter %}</button>
```
Comment (emkll) [inline securedrop/journalist_templates/edit_account.html]: Will this work with all locales/languages? We should explicitly test this as part of QA

**Comment by eloquence** [comment]
(Hold on merge for one follow-up commit, we agreed to use `text-transform: uppercase` where appropriate, consistent w/ previous discussion in #1587.)

**Comment by eloquence** [comment]
`.upper` class for using `text-transform:uppercase` on strings that would otherwise be touched in this PR introduced in 0cc9bc7740243a2c85411040d3b6530bead37130.
